internal/vpn/client.go

```diff
@@ -112,26 +113,48 @@ func (c *Client) Serve() error {
 		return fmt.Errorf("error setting up direct routes: %w", err)
 	}
 
+	if runtime.GOOS == "windows" {
+		// okay, so, here's done because after the `SetupTUN` call,
+		// interface doesn't get its values immediately. Reason is unknown,
+		// all credits go to Microsoft. Delay may be different, this one is
+		// fairly large to cover not really performant systems.
+		time.Sleep(10 * time.Second)
+	}
+
 	defer c.routeTrafficDirectly(tunGateway)
 	c.log.Infof("Routing all traffic through TUN %s", tun.Name())
 	if err := c.routeTrafficThroughTUN(tunGateway); err != nil {
 		return fmt.Errorf("error routing traffic through TUN %s: %w", tun.Name(), err)
 	}
 
+	rw := io.ReadWriter(c.conn)
+	if encrypt {
+		c.log.Infoln("Enabling encryption...")
+
+		rw, err = WrapRWWithNoise(c.conn, true, c.cfg.Credentials.PK, c.cfg.Credentials.SK)
+		if err != nil {
+			return fmt.Errorf("failed to enable encryption: %w", err)
+		}
+
+		c.log.Infoln("Encryption enabled")
+	} else {
+		c.log.Infoln("Encryption disabled")
+	}
+
 	connToTunDoneCh := make(chan struct{})
 	tunToConnCh := make(chan struct{})
```

Comment:
Instead of using 2 channels here, I would use one channel and a `sync.Once` here, but I guess that is personal taste.

Reply:
Personally I don't like these 2 channels much, but `sync.Once` due to its syntax may just produce more code. I'd stick to 2 channels in this case